bigtable/client/internal/table_admin.cc

```diff
@@ -188,6 +188,33 @@ bool TableAdmin::CheckConsistency(
   return response.consistent();
 }
 
+bool TableAdmin::WaitForConsistencyCheckImpl(
+    bigtable::TableId const& table_id,
+    bigtable::ConsistencyToken const& consistency_token,
+    std::unique_ptr<bigtable::PollingPolicy> polling_policy) {
+  btproto::CheckConsistencyRequest request;
+  request.set_name(TableName(table_id.get()));
+  request.set_consistency_token(consistency_token.get());
+  MetadataUpdatePolicy metadata_update_policy(
+      instance_name(), MetadataParamTypes::NAME, table_id.get());
+
+  grpc::Status status;
+  do {
+    auto response = ClientUtils::MakeCall(
+        *client_, rpc_retry_policy_->clone(), rpc_backoff_policy_->clone(),
+        metadata_update_policy, &AdminClient::CheckConsistency, request,
+        "CheckConsistency", status, true);
+
+    if (status.ok()) {
+      return true;
```

Comment:
I think you need to look at the value of `response` right?  The way this works is that we are waiting for one more more clusters to catch up to the point represented by the consistency token.  The first N times we call it may return "not consistent yet" and we need to continue waiting.